src/meson.build

```diff
@@ -0,0 +1,85 @@
+src_cpp = files(
+  'ast.cpp',
+  'ast2c.cpp',
+  'ast_fwd_decl.cpp',
+  'ast_sel_cmp.cpp',
+  'ast_sel_super.cpp',
+  'ast_sel_unify.cpp',
+  'ast_sel_weave.cpp',
+  'ast_selectors.cpp',
+  'ast_supports.cpp',
+  'ast_values.cpp',
+  'backtrace.cpp',
+  'base64vlq.cpp',
+  'bind.cpp',
+  'c2ast.cpp',
+  'check_nesting.cpp',
+  'color_maps.cpp',
+  'constants.cpp',
+  'context.cpp',
+  'cssize.cpp',
+  'emitter.cpp',
+  'environment.cpp',
+  'error_handling.cpp',
+  'eval.cpp',
+  'eval_selectors.cpp',
+  'expand.cpp',
+  'extender.cpp',
+  'extension.cpp',
+  'file.cpp',
+  'fn_colors.cpp',
+  'fn_lists.cpp',
+  'fn_maps.cpp',
+  'fn_miscs.cpp',
+  'fn_numbers.cpp',
+  'fn_selectors.cpp',
+  'fn_strings.cpp',
+  'fn_utils.cpp',
+  'inspect.cpp',
+  'json.cpp',
+  'lexer.cpp',
+  'listize.cpp',
+  'operators.cpp',
+  'output.cpp',
+  'parser.cpp',
+  'parser_selectors.cpp',
+  'plugins.cpp',
+  'position.cpp',
+  'prelexer.cpp',
+  'remove_placeholders.cpp',
+  'sass.cpp',
+  'sass2scss.cpp',
+  'sass_context.cpp',
+  'sass_functions.cpp',
+  'sass_values.cpp',
+  'source.cpp',
+  'source_map.cpp',
+  'stylesheet.cpp',
+  'to_value.cpp',
+  'units.cpp',
+  'utf8_string.cpp',
+  'util.cpp',
+  'util_string.cpp',
+  'values.cpp'
+)
+src_c = files(
+  'c99func.c',
+  'cencode.c'
+)
```

Comment:
Would it be possible to have the above in one separate file and include it here?
I would like to keep the necessary file to generate to a minimum.

Reply:
Meson doesn't support includes from about anywhere, it can only do subdirectory inclusion. However we can merge `src_cpp` and `src_c` into one variable and have the rest of the file be a template.

Reply:
Hmm, that seems rather strange to me, but you seem correct: https://github.com/mesonbuild/meson/pull/3557. Merging the variables was not the point. I just want to avoid having any logic being in the template and generated file.

Since the file in question seems rather small it's fine for me for now.

Reply:
> I just want to avoid having any logic being in the template and generated file.

I don't understand how any kind of logic would make its way onto the template here. It's just "replace the placeholder with the list of sources" and having an inclusion feature in Meson would do nothing more than just "open the file and replace its contents".

Alternatively we could have scripts run by Meson at configure or at build time if there's really a need to have conditional generation not handled by Meson directly. But I don't think that's needed here.

Reply:
As I said, I'm fine for the moment how it is.

The point is similar to MSVC, here the logic is in https://github.com/mgreter/libsass/blob/libsass-refactor-snapshot/win/libsass.vcxproj#L190. I only generate a small file that lists all sources and that file get imported into the the actual project file. All that logic lives in the repo and not in the template. I hope that clears it up a bit why I ideally want to separate them.

Reply:
Ah, I see. I thought by logic you mean having to parse and execute expression from within the template file.